src/components/UI/Dropdown.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import { faChevronUp } from '@fortawesome/free-solid-svg-icons';
+
+import classes from './Dropdown.module.scss';
+
+type Dropdownprops = {
+  header: string;
+  options?: string[];
+};
+
+function Dropdown({ header }: Dropdownprops) {
+  const [open, setOpen] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  const toggleWindow = () => setOpen(!open);
+  const selectItem = () => setOpen(false);
+
+  useEffect(() => {
+    let handler = (e: any) => {
+      if (!ref.current?.contains(e.target as Node)) {
+        setOpen(false);

```

Comment:
A lot of duplicated code, 2 functions - setOpen and separate setOpen

Why e: any ?